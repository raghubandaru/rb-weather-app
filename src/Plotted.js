import React from "react";
import Plot from "react-plotly.js";
import { toJS } from "immutable";

class Plotted extends React.Component {
  shouldComponentUpdate(nextProps) {
    const xDataChanged = !this.props.xData.equals(nextProps.xData);
    const yDataChanged = !this.props.yData.equals(nextProps.yData);

    return xDataChanged || yDataChanged;
  }

  componentDidMount() {
    console.log("component mounted");
  }

  componentDidUpdate() {
    console.log("component updated");
  }

  render() {
    console.log("RENDER PLOT");
    let data = [
        {
          x: this.props.xData.toJS(),
          y: this.props.yData.toJS(),
          type: "scatter",
          marker: { color: "red" }
        }
      ],
      layout = {
        margin: {
          t: 10,
          r: 50,
          l: 50
        }
      },
      config = { displayModeBar: false };
    return (
      <Plot
        data={data}
        layout={layout}
        config={config}
        className="plo"
        useResizeHandler={true}
        onClick={this.props.onPlotClick}
      />
    );
  }
}

export default Plotted;
