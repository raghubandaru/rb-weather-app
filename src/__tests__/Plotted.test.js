import React from "react";
import { shallow } from "enzyme";
import { fromJS } from "immutable";
import Plotted from "../Plotted";

describe("<Plotted />", function() {
  it("renders correctly", function() {
    const wrapper = shallow(
      <Plotted
        xData={fromJS({})}
        yData={fromJS({})}
        onClick={() => console.log("clicked on the plot")}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
