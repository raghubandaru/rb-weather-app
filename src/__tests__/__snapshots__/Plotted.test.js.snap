// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Plotted /> renders correctly 1`] = `
<PlotlyComponent
  className="plo"
  config={
    Object {
      "displayModeBar": false,
    }
  }
  data={
    Array [
      Object {
        "marker": Object {
          "color": "red",
        },
        "type": "scatter",
        "x": Object {},
        "y": Object {},
      },
    ]
  }
  debug={false}
  layout={
    Object {
      "margin": Object {
        "l": 50,
        "r": 50,
        "t": 10,
      },
    }
  }
  style={
    Object {
      "display": "inline-block",
      "position": "relative",
    }
  }
  useResizeHandler={true}
/>
`;
